bug_1292234.patch

```diff
@@ -306,6 +306,111 @@ diff --git a/devtools/server/actors/storage.js b/devtools/server/actors/storage.
  StorageActors.createActor({
    typeName: "Cache",
  }, {
+diff --git a/devtools/server/tests/browser/browser.ini b/devtools/server/tests/browser/browser.ini
+--- a/devtools/server/tests/browser/browser.ini
++++ b/devtools/server/tests/browser/browser.ini
+@@ -122,6 +122,7 @@ skip-if = e10s # Bug 1183605 - devtools/
+ [browser_storage_listings.js]
+ [browser_storage_updates.js]
+ skip-if = (verify && debug && (os == 'mac' || os == 'linux'))
++[browser_storage_webext_storage_local.js]
+ [browser_stylesheets_getTextEmpty.js]
+ [browser_stylesheets_nested-iframes.js]
+ [browser_register_actor.js]
+diff --git a/devtools/server/tests/browser/browser_storage_webext_storage_local.js b/devtools/server/tests/browser/browser_storage_webext_storage_local.js
+new file mode 100644
+--- /dev/null
++++ b/devtools/server/tests/browser/browser_storage_webext_storage_local.js
+@@ -0,0 +1,89 @@
++/* vim: set ft=javascript ts=2 et sw=2 tw=80: */
++/* Any copyright is dedicated to the Public Domain.
++   http://creativecommons.org/publicdomain/zero/1.0/ */
++
++/* import-globals-from storage-helpers.js */
++
++"use strict";
++
++Services.scriptloader.loadSubScript(
++  "chrome://mochitests/content/browser/devtools/server/tests/browser/storage-helpers.js",
++  this);
++
++const {Toolbox} = require("devtools/client/framework/toolbox");
++
++async function setupExtensionDebuggingToolbox(extensionId, {openToolbox} = {}) {
++  DebuggerServer.init();
++  DebuggerServer.registerAllActors();
++  const transport = DebuggerServer.connectPipe();
++  const client = new DebuggerClient(transport);
++  await client.connect();
++  const front = await client.mainRoot.getAddon({id: extensionId});
++  const target = await front.connect();
++
++  const result = {target, front};
++
++  if (openToolbox) {
++    result.toolbox = await gDevTools.showToolbox(target, null, Toolbox.HostType.WINDOW);
++  }
++
++  return result;
++}
++
++add_task(async function() {
++  async function background() {
++    /* globals browser */
++    browser.test.onMessage.addListener(async (msg, ...args) => {
++      switch (msg) {
++        case "storage-local-set":
++          await browser.storage.local.set(args[0]);
++          break;
++        default:
++          browser.test.fail(`Unexpected test message: ${msg}`);
++      }
++
++      browser.test.sendMessage(`${msg}:done`);
++    });
++
++    browser.test.sendMessage("extension-origin", window.location.origin);
++  }
++  const extension = ExtensionTestUtils.loadExtension({
```

Comment:
> This is found here: https://searchfox.org/mozilla-central/source/testing/mochitest/tests/SimpleTest/ExtensionTestUtils.js ; this is for mochitests. In every mochitest, you get that instance of ExtensionTestUtils.

> But there is a different location for XPCShell tests: https://searchfox.org/mozilla-central/source/toolkit/components/extensions/ExtensionXPCShellUtils.jsm. This must be explicitly included in our xpcshell tests. It's not automatically available. We have to import the test utils.

> In this directory, https://searchfox.org/mozilla-central/source/toolkit/components/extensions/test/xpcshell, these are all the xpcshell tests for extensions (mostly). Since this is in toolkit, we can run on different applications like Fennec (mobile).

> https://searchfox.org/mozilla-central/source/browser/components/extensions/test/xpcshell/ this is in browser level, so this only runs on Firefox Desktop.

> The way we ensure that the two ExtensionTestUtils defined separately work mostly the same is because they share this code: https://searchfox.org/mozilla-central/source/toolkit/components/extensions/ExtensionTestCommon.jsm . You don't use it directly.